src/main/resources/liquibase/crop_changelog/21_3_0.xml

```diff
@@ -0,0 +1,814 @@
+			</sqlCheck>
+		</preConditions>
+		<comment>
+			Remove 'Germplasm Descriptor' type to SEED_SOURCE
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			DELETE FROM cvtermprop WHERE cvterm_id = 8360 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="v21.3.0-19">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">
+				SELECT count(1) FROM cvtermprop WHERE cvterm_id = 8360 AND type_id = 1800;
+			</sqlCheck>
+		</preConditions>
+		<comment>
+			Remove SEED_SOURCE variable from cvterm table and from cvterm_relationship table
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			delete from cvterm_relationship where subject_id = 8360;
+			delete from cvtermprop where cvterm_id = 8360;
+			delete from variable_overrides where cvterm_id = 8360;
+			delete from cvtermsynonym where cvterm_id = 8360;
+			delete from cvterm where cvterm_id = 8360;
+		</sql>
+	</changeSet>
+	<!-- End of SEED_SOURCE related changesets-->
+
+	<!-- ENTRY_TYPE related changesets-->
+	<changeSet author="lkovacic" id="v21.3.0-20">
+		<preConditions onFail="MARK_RAN">
+				<sqlCheck expectedResult="1">
+					SELECT count(1) FROM cvtermprop WHERE cvterm_id = 8255 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+				</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_TYPE' variable type to entry detail</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE cvtermprop SET value = 'Entry Detail' WHERE cvterm_id = 8255 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+
+			INSERT INTO entry_detail_migration_output (cvterm_id) VALUES (8255);
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-21">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT count(1) > 0 FROM projectprop WHERE variable_id = 8255 AND type_id = 1804;
+			</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_TYPE' variable type to entry detail in 'projectprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE projectprop SET type_id = 1815 WHERE variable_id = 8255 AND type_id = 1804;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-22">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="stockprop" columnName="cvalue_id"/>
+			</not>
+		</preConditions>
+		<comment>Add column 'cvalue_id' to 'stockprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			ALTER TABLE stockprop ADD cvalue_id INT NULL;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-23">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<columnExists tableName="stockprop" columnName="cvalue_id"/>
+				<sqlCheck expectedResult="0">
+					SELECT count(1) FROM stockprop WHERE cvalue_id IS NOT NULL;
+				</sqlCheck>
+				<sqlCheck expectedResult="1">
+					SELECT count(*) > 0
```

Comment:
@lik-83 you can remove a select here using 
``
SELECT COUNT(DISTINCT( sp.type_id )) > 0 AS typeId
						  FROM stockprop sp
								   INNER JOIN cvterm_relationship cr
						  ON sp.type_id = cr.subject_id
							  AND cr.type_id = 1220
								   INNER JOIN cvterm_relationship cr2
						  ON cr.object_id = cr2.subject_id
							  AND cr2.type_id = 1190
``

Reply:
Done